big-practice/src/js/pages/dashboard.page.js

```diff
@@ -131,41 +141,61 @@ class Dashboard {
     }
   }
 
-  // /**
-  //  * Handle update form
-  //  */
-  // handleUpdateForm() {
-  //   const data = {
-  //     name: nameMovieInput.value,
-  //     director: directorInput.value,
-  //     nation: nationInput.value,
-  //   };
-  //   const config = {
-  //     name: ['empty'],
-  //     director: ['empty'],
-  //     nation: ['empty'],
-  //   };
-  //   const validate = validateForm(data, config);
-  //   const formMovieId = form.getAttribute('data-id');
-
-  //   if (!validate.isValid) {
-  //     showErrorMessage(nameMovieInput, validate.errors.name);
-  //     showErrorMessage(directorInput, validate.errors.director);
-  //     showErrorMessage(nationInput, validate.errors.nation);
-  //     return;
-  //   }
-
-  //   getApi(`${MOVIES_API}?name=${data.name}`, (movieList) => {
-  //     const moviesDoNotExist = movieList.length === 0 || movieList[0].id === parseInt(formMovieId, 10);
-
-  //     if (moviesDoNotExist) {
-  //       putApi(`${MOVIES_API}/${formMovieId}`, data, () => { renderTable(); });
-  //       hideModal(modalForm);
-  //     } else {
-  //       showErrorMessage(nameMovieInput, MESSAGES.exist);
-  //     }
-  //   });
-  // }
+  /**
+   * Handle update form by calling API
+   */
+  async handleUpdateForm() {
+    try {
+      const data = {
+        name: this.nameMovieInput.value,
+        director: this.directorInput.value,
+        nation: this.nationInput.value,
+      };
+      const config = {
+        name: ['empty'],
+        director: ['empty'],
+        nation: ['empty'],
+      };
+      const validate = formValidate.validateForm(data, config);
+      const formMovieId = this.form.getAttribute('data-id');
+
+      if (!validate.isValid) {
+        DocumentHelper.showErrorMessage(this.nameMovieInput, validate.errors.name);
+        DocumentHelper.showErrorMessage(this.directorInput, validate.errors.director);
+        DocumentHelper.showErrorMessage(this.nationInput, validate.errors.nation);
+
+        return;
+      }
+
+      const movieList = await apiService.get(`${MOVIES_API}?name=${data.name}`);
+      const moviesDoNotExist = movieList.length === 0
```

Comment:
moviesDoNotExist does not make sense. Boolean variable should start with is, has or enable/disable. Ex: isValidUser

Reply:
Fixed